src/Microsoft.AspNetCore.DataProtection/Repositories/FileSystemXmlRepository.cs

```diff
@@ -191,6 +192,12 @@ public virtual void StoreElement(XElement element, string friendlyName)
             }
 
             StoreElementCore(element, friendlyName);
+
+            if (DockerUtils.IsDocker && !DockerUtils.IsVolumeMountedFolder(Directory))
```

Comment:
Why are we logging here vs https://github.com/aspnet/DataProtection/blob/c959795a6461a9150fbabfd5c08486d05e363057/src/Microsoft.AspNetCore.DataProtection/KeyManagement/XmlKeyManager.cs#L456 . I don't think user cares about warning message if he configured directory himself. Also having it on save is a bit undiscoverable, if key already exists user won't see the warning until it expires.